libp2p/protocols/pubsub/floodsub.nim

```diff
@@ -177,7 +177,7 @@ method rpcHandler*(
   f.updateMetrics(rpcMsg)
 
 method init*(f: FloodSub) =
-  proc handler(conn: Connection, proto: string) {.async.} =
+  proc handler(conn: Connection, proto: string) {.async: (raises: [CancelledError]).} =
     ## main protocol handler that gets triggered on every
     ## connection for a protocol string
     ## e.g. ``/floodsub/1.0.0``, etc...
```

Comment:
here `CancelledError` has to be raised because it is handler, but this handler is catching this exception.

Reply:
it might be a good idea to review whether the handler _should_ catch the cancellation or let it propagate - much of this code was written as cancellation was being developed and it might not always be following best practice .. catching the cancellation and translating it to some other outcome is sometimes the right thing, but often not (it "hides" the fact that a cancellation is ongoing from the calling code which deprives it of the means to make the distinction).

Reply:
hey @arnetheduck, yes, it makes much more sense to catch cancellation. this effort is in #1260